core/src/main/java/com/linecorp/armeria/server/HttpServiceOptionsBuilder.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+
+/**
+ * Creates a new {@link HttpServiceOptions} with the specified parameters.
+ */
+public final class HttpServiceOptionsBuilder {
+    private long requestTimeoutMillis = -1;
+    private long maxRequestLength = -1;
+    private long requestAutoAbortDelayMillis = -1;
+
+    HttpServiceOptionsBuilder() {}
+
+    /**
+     * Returns the server-side timeout of a request in milliseconds.
+     */
+    public HttpServiceOptionsBuilder requestTimeoutMillis(long requestTimeoutMillis) {
+        this.requestTimeoutMillis = requestTimeoutMillis;
```

Comment:
We need to check the value:
```
checkArgument(requestTimeoutMillis >= 0,
              "requestTimeoutMillis: %s (expected: >= 0)", requestTimeoutMillis);
```
After that, we also need to modify these lines:
https://github.com/line/armeria/pull/5574/files#diff-2246967d4d345f650f2b3ec5d5032040fe494cc032094448c9c05c8557665449R188-R190

Reply:
Thank you for checking. I've added the validation logic and applied the changes to the specified code that you mentioned.